# Variables — exercises


Work in the answer cells below. Predict an answer before running code, then
compare the result with your prediction. The code cells provide starting
values, not completed solutions. Deliberately faulty code is shown as text;
copy it into an answer cell when you are ready to repair it.

Every complete exercise session has **40 trials**. `1` means correct and `0`
means incorrect. The exercise training rule is **accuracy > 0.66** (no 0.75 as in the lectures). Because
one response is 1/40 = 0.025 of the session, 26/40 = 0.65 does not qualify,
whereas 27/40 = 0.675 does.

## Exercise 1: make the calculation readable

An animal called Milo answered 29 of 40 trials correctly. This calculation
works, but the two numbers do not explain themselves:

In [1]:
29 / 40

0.725

Store the animal's name as `animal_name`, the correct count as `n_correct`,
and the total count as `n_trials`. Calculate `accuracy` from the named values.
Display the name and accuracy. Explain why the name needs quotes.

**Check after writing your code:** 29/40 is 0.725. Then use 26 and 27 correct
responses. Predict both decisions under `accuracy > 0.66` before calculating.

In [2]:
# Your code here.

## Exercise 2: put two blocks together

The same session consists of two blocks of 20 trials. The animal was correct
15 times in the first block and 12 times in the second.

Calculate and display:

1. The total correct count summed over both blocks, `n_correct`.
2. The incorrect count (trials count - correct count), `n_incorrect`.
3. The accuracy, `accuracy` = correct count / trial count.
4. The percentage correct, `percent_correct` = accuracy × 100.

Reuse your calculated values. Check that correct and incorrect counts add to 40.

In [3]:
first_block_correct = 15
second_block_correct = 12
n_trials = 40
# Your calculations here.

## Exercise 3: compare two sessions

Nori completed one 40-trial session on each of two days. It answered 23 trials
correctly on the first day and 28 on the second day.

Calculate each day's accuracy and how many more correct responses occurred
on the second day. Use descriptive names and print the results.

In which session did the animal perform better? Which session meets the
exercise rule `accuracy > 0.66`?

In [ ]:
first_day_correct = 23
second_day_correct = 28
n_trials = 40
# Your code here.

## Exercise 4: choose the needed type conversion

Given the variables below, write:

1. One expression that produces the **number** `9`.
2. One expression that produces the **text** `"72"`.

Use `int(...)` or `str(...)` where needed. Predict the type of each result,
then print each result and its type.

Finally, convert `count_as_text` to a number and calculate its accuracy for
a 40-trial session.

In [ ]:
x = "7"
y = 2
count_as_text = "27"
n_trials = 40
# Write the two expressions, then convert the count and calculate accuracy.